### PR TITLE
added authentication, firebase storage and firebase database usage

### PR DESCRIPTION
**Pull request by HasaniH**

### CODE REVIEW

File: android/cloud_spin_v2/.gitignore [hunk 1]
```diff
@@ -7,3 +7,6 @@
 /build
 /captures
 .externalNativeBuild
+
```
Comment (mimming) [inline android/cloud_spin_v2/.gitignore]: Add a license header to this file

File: android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java [hunk 2]
```diff
@@ -69,6 +70,7 @@ protected void onCreate(Bundle savedInstanceState) {
         FirebaseAuth auth = FirebaseAuth.getInstance();
         if (auth.getCurrentUser() != null) {
             // already signed in
```
Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java]: Be consistent in comment formatting.    Do them in sentence case, with a leading space per https://source.android.com/source/code-style

(Apply this to all of the inline comments in the project)

File: android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java [hunk 3]
```diff
@@ -69,6 +70,7 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            Log.v("hello", "world");
```
Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java]: Either replace this logging message with something more useful, or add a // TODO comment explaining what to follow up with per https://source.android.com/source/code-style#use-todo-comments

File: android/cloud_spin_v2/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -26,13 +26,12 @@ Copyright 2017 Google Inc.
         android:roundIcon="@mipmap/ic_launcher_round"
```
Comment (mimming) [inline android/cloud_spin_v2/app/src/main/AndroidManifest.xml]: The app should probably be locked in portrait mode, consider achieving that by adding this attribute to the applcation - android:screenOrientation="portrait"

File: android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java [hunk 5]
```diff
@@ -12,13 +12,14 @@
...
 
 import android.content.Intent;
 import android.net.Uri;
 import android.provider.MediaStore;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.View;
 import android.widget.Button;
 import android.widget.Toast;
```
Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java]: Down on line 40 to 53 - your class field variables of your activity have some issues.

Style: All of them should follow the naming convention in the style guidelines (start with a lowercase m) - https://source.android.com/source/code-style#follow-field-naming-conventions

Visibility: If you don't specify public or private, they're package scope.  This is rarely what you want.  Unless you have a good reason for them to be package scope, make them private.

Unnecessary static - a bunch of them are static when they don't need to be (and shouldn't be).  databaseClips, for example.    Only declare them static if you're sure you need to use them from a static context (which isn't the case for any of these)

Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java]: On line 48, there is unnecessary whitespace between super. and onCreate.  Remove it.

Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java]: On lines 40 to 53 - your class field variables use both camelCase and snake_case.  Be consistent: use camelCase per the Android style guidelines.

File: android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java [hunk 6]
```diff
@@ -69,6 +70,7 @@ protected void onCreate(Bundle savedInstanceState) {
         FirebaseAuth auth = FirebaseAuth.getInstance();
```
Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java]: Up on line 60 - You use a CamelCase name for a path in Firebase - getReference("VideoClips");

Firebase paths should be all lowercase, with dashes delimiting words.  Use `video-clips` instead.

File: android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/VideoClip.java [hunk 7]
```diff
@@ -1,6 +1,4 @@
-package com.cloud_spin.hasanih.cloud_spin_v2;
-
-import android.provider.MediaStore;
+package com.google.cloudspinphoto;
```
Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/VideoClip.java]: Add a license header to this file, and all other source files.

File: android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/VideoClip.java [hunk 8]
```diff
@@ -1,6 +1,4 @@
...
 
 import com.google.firebase.database.DatabaseReference;
```
Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/VideoClip.java]: Remove the 'created by' JavaDoc comment generated by your IDE

Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/VideoClip.java]: For POJOs, use the IDE's code generation to fully generate getters and setters.

Also don't squish them on to one line.  Add newlines just like you would for any other Java class.

File: android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java [hunk 9]
```diff
@@ -69,6 +70,7 @@ protected void onCreate(Bundle savedInstanceState) {
...
         if (auth.getCurrentUser() != null) {
             // already signed in
+            Log.v("hello", "world");
         } else {
             // not signed in
             startActivityForResult(
```
Comment (mimming) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java]: On line 87, why is dispatchTakeVideoIntent public?  It should probably be private?

On line 96, why is addClipToDatabase protected? It should probably also be private

File: android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java [hunk 10]
```diff
@@ -41,71 +41,70 @@
...
+    // Grab a database reference object
+    private DatabaseReference mDatabaseClips;
+    private String mId;
+    private VideoView mResultVideo;
 
-    static final int REQUEST_VIDEO_CAPTURE = 1;
-    private static final int RC_SIGN_IN = 123;
-
-    VideoView result_video;
+    private final int mREQUEST_VIDEO_CAPTURE = 1;
```
Comment (HasaniH) [inline android/cloud_spin_v2/app/src/main/java/com/google/cloudspinphoto/MainActivity.java]: I actually made the constants public and removed the leading "m." Everywhere the constants are called, Ignore the leading "m"

File: android/cloud_spin_v2/.gitignore [hunk 11]
```diff
@@ -1,3 +1,17 @@
+// Copyright 2017 Google Inc.
```
Comment (mimming) [inline android/cloud_spin_v2/.gitignore]: .gitignore comments should start with a #, not a /

See https://stackoverflow.com/questions/8865848/comments-in-gitignore or run `$ git help gitignore` to learn more.
